pulp_rpm/app/models/modulemd.py

```diff
@@ -44,6 +44,7 @@ class Modulemd(Content):
     context = models.CharField(max_length=255)
     arch = models.CharField(max_length=255)
 
+    static_context = models.BooleanField(default=False)
```

Comment:
I feel like we might need to make this nullable with a null default for compatibility reasons. @goosemania wdyt?

Reply:
Yeah... I have the same feeling but I'm not sure :)

I think we need to publish modules as is, old modules should be published without static_context , new ones which have it, should be published with it. 
At the same time I'm confused by the mandatory nature of static_context according to [the spec](https://github.com/fedora-modularity/libmodulemd/commit/2000e88d48a7b8d9fba1f7866d3709035dd2b957#diff-f693f5c1bd57e7782999ed7d59a4e2cdb1548c6d5e2e4663265f2165a949effeR81-R96).

This leads to more questions, e.g. is it allowed to have mixed modules in one repo (with static_context and without, basically the current approach with the fixtures repo).
Maybe we need to ask modularity folks and/or dnf.

Thinking more about all that... regardless of what is in the db, we'll publish modules as is by default, so old modules will preserve their format. We use yaml snippets and we do not regenerate yaml based on the data in the db. So... maybe we do not need null value at all.

The only use case for the null value that I can imagine is that we need to know the original format of a module.

Reply:
I'm fine with either. It feels safe to not have a default at the model level.

Reply:
I think we should bring this up with the modularity folks, but it seems obvious that it can't be "mandatory" since all RHEL 8 metadata in existence will essentially not have this.

Maybe what they mean is that it's mandatory to be able to retrieve this value (ie. get a default) although I'm not convinced of this.  I think "mandatory" is just a mistake.

+1 to "no default", personally.  But we still need to treat a null value as if it were false in the places where it's relevant.

Reply:
I'm asking on IRC ^

Reply:
Changed to allow null